src/components/PokemonDetailPage/PokemonDetailPage.js

```diff
@@ -0,0 +1,149 @@
+import React ,{useEffect} from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import axios from 'axios';
+import Title from '../Title/Title';
+import styled from 'styled-components';
+import image from '../../backgrounds.jpg';
+import { Link } from 'react-router-dom';
+
+const StyleDiv=styled.div`
+    background-image:url(${image});
+    background-repeat: no-repeat;
+    background-attachment: fixed;
+    background-size: cover;
+    min-height:100%;
+    min-width:100%;
+    position:absolute;
+
+    ul li{
+        text-align:left;
+        margin-left:42%;
+    }
+`;
+
+const NameDiv=styled.div`
+    padding-top:80px;
+    color:#000066;
+    font-size:36px;
+    text-align:center;
+    text-transform:uppercase;
+    background-color:#00ccff;
+    border-radius:25%;
+    width:30%;
+    margin-left:35%;
+`;
+
+const DataDiv=styled.div`
+    background-color:#ffffff;
+    text-align:center;
+    width:40%;
+    border:4px solid #00ccff;
+    border-radius:25%;
+    margin-top:3%;
+    margin-left:30%;
+    margin-right:30%;
+    overflow:hidden;
+`;
+
+const LabelDiv=styled.div`
+    color:white;
+    background-color:#000066;
+`;
+
+const TypeBadge=styled.div`
+    border-radius:50px 50px 50px 50px;
+    margin-left:40%;
+    width:20%;
+    padding-top:3%;
+    padding-bottom:3%;
+    margin-top:1%;
+    margin-bottom:1%;
+    ${(props)=>(props.type==='water'&&`color:white; background-color:#0000ff;`)||
```

Comment:
This is a bit too overwhelming, but we appreciate that you tried to create an unique badge for each type.